File: apps/ios/ScribbleFit/LocalPackages/Features/FeatureWorkouts/Sources/FeatureWorkouts/Domain/UseCases/FormatWorkoutSummaryUseCase.swift
import Foundation
import CoreModel

public struct VolumeSummary: Sendable {
    public let value: String
    public let isKilo: Bool
    
    public init(value: String, isKilo: Bool) {
        self.value = value
        self.isKilo = isKilo
    }
}

public struct FormatWorkoutSummaryUseCase: Sendable {
    public init() {}
    
    public func execute(totalVolume: Double) -> VolumeSummary {
        if totalVolume >= 1000 {
            return VolumeSummary(
                value: String(format: "%.1f", totalVolume / 1000.0),
                isKilo: true
            )
        } else {
            return VolumeSummary(
                value: String(format: "%.0f", totalVolume),
                isKilo: false
            )
        }
    }
}

File: apps/android/feature/workouts/domain/src/main/java/com/scribblefit/feature/workouts/domain/usecase/FormatWorkoutSummaryUseCase.kt
package com.scribblefit.feature.workouts.domain.usecase

class FormatWorkoutSummaryUseCase {
    operator fun invoke(totalVolume: Double): VolumeSummary {
        return if (totalVolume >= 1000) {
            VolumeSummary(
                value = "%.1f".format(totalVolume / 1000.0),
                isKilo = true
            )
        } else {
            VolumeSummary(
                value = "%.0f".format(totalVolume),
                isKilo = false
            )
        }
    }

    data class VolumeSummary(
        val value: String,
        val isKilo: Boolean
    )
}
